staking/slash/interface.go

```diff
@@ -0,0 +1,24 @@
+package slash
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/harmony-one/harmony/core/types"
+	"github.com/harmony-one/harmony/internal/params"
+	"github.com/harmony-one/harmony/shard"
+	staking "github.com/harmony-one/harmony/staking/types"
+)
+
+// CommitteeReader ..
+type CommitteeReader interface {
+	Config() *params.ChainConfig
+	ReadShardState(epoch *big.Int) (*shard.State, error)
+	CurrentBlock() *types.Block
+}
+
+// stateDB is the abstracted interface for *state.DB
+type stateDB interface {
```

Comment:
I see why no longer ptr

Reply:
Yep! I am using it as a simple adapter between modules that can help with decreasing the correlation between modules and specifies a clean boundary.

Reply:
I prefer no interface on this , the other interfaces you see are for breaking import cycle but this is an interface for convenience for you. In general we should use interfaces only when we hit an import cycle or when we have several concrete types and need to abstract over it. Otherwise it is not worth the complexity and runtime hit

Reply:
No interface now. Done.